gemviz/mainwindow.py

```diff
@@ -88,15 +74,70 @@ def doOpen(self, *args, **kw):
         previous_uri = settings.getKey(TILED_SERVER_SETTINGS_KEY) or ""
         server_uri = TiledServerDialog.getServer(self)
         if server_uri is None:
-            self.status = "No tiled server selected."
+            self.setStatus("No tiled server selected.")
             return
-        self.status = f"selected tiled {server_uri=!r}"
+        self.setStatus(f"selected tiled {server_uri=!r}")
 
         try:
             client = utils.connect_tiled_server(server_uri)
         except Exception as exc:
-            self.status = f"Error for {server_uri=!r}: {exc}"
+            self.setStatus(f"Error for {server_uri=!r}: {exc}")
             settings.setKey(TILED_SERVER_SETTINGS_KEY, previous_uri)
             return
 
-        self.filter_panel.setServer(client)
+        self.setServer(server_uri, client)
+
+    def catalog(self):
```

Comment:
there is a `catalog` method but no `setCatalog` method? From what I can tell what plays the role of `setCatalog` is  the `catalogSelected` method below, where `self._catalog = self.server()[catalog_name]`. However the `_catalogSelected` attribute is defined as `self._catalogSelected = catalog_name`, which seems a bit inconsistent to me (i.e. `catalogSelected` fulfill a very different function than its internal counterpart `_catalogSelected`). Besides I am confused by the fact that there is 2 catalogSelected method, one in `mainwindow.py` and one in `bluesky_runs_catalog_search.py`. Would it make more sense to rename the `catalogSelected` method below `setCatalog`? Sorry if I am not clear, I have not digested the whole thing just yet...

Reply:
This is what I understand (I writing it done mostly to help me process it...).

**In `mainwindow.py`** `catalogSelected` is called when `self.catalogs.currentTextChanged.connect(self.catalogSelected)`:
- it checks if the catalog selected name, i.e. `self._catalogSelected` is valid
- it checks if the catalog selected, i.e. `self.catalog`, is a BRC
- if so, it instantiate `self.mvc_catalog = BRC_MVC(self)` which basically populate the app with the correct widgets.

**In `bluesky_runs_catalog.py`** `catalogSelected` is called when `self.brc_search_panel.catalogSelected(self.catalog().item["id"])`:
- it checks if the catalog selected name, i.e. `self._catalogSelected` is valid (which is already done in `mainwindows`?)
- it checks that the catalog name is not an empty string
- it looks sets the date_time_widget boundaries

Reply:
You are correct that the code is confusing.  This part was designed early.  Refactor needed...

* mainwindow's `setCatalog()` should be the slot to received the pop-up menu selection, the value is a catalog **name**.
* bluesky_runs_catalog's `catalogSelected()` only needs to set the date widget, validity check is redundant and can be removed, its name should be changed to reflect what it does such as `setupCatalog()`
* the date_time_widget boundaries (`Min`, `Max`, `Low`, `High`) should be set in a single call  to the date_time_widget which will be responsible for updating its attributes.